### PR TITLE
Allow function rebalance to be run periodically

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

After a rebalance mechanism was implement for pulsar functions in:

https://github.com/apache/pulsar/pull/7388

it is only natural to add the functionality for the rebalance to be triggered periodically.

### Modifications

Allow rebalance to be triggered on a user defined frequency

### CODE REVIEW

File: conf/functions_worker.yml [hunk 1]
```diff
@@ -93,6 +93,8 @@ schedulerClassName: "org.apache.pulsar.functions.worker.scheduler.RoundRobinSche
 functionAssignmentTopicName: "assignments"
 failureCheckFreqMs: 30000
 rescheduleTimeoutMs: 60000
+# frequency at which to check if cluster needs rebalancing (15min default interval)
+rebalanceCheckFreqSec: 900
```
Comment (srkukarni) [inline conf/functions_worker.yml:97]: Do we want to turn it on by default? Also is 15 min too frequent

**Comment by wolfstudy** [comment]
@jerrypeng can we update the **Motivation** and **Modifications**?

File: conf/functions_worker.yml (hunk 1 above)
Comment (jerrypeng) [reply conf/functions_worker.yml:97]: Set it 1 hour.  Is there any harm to enable by default?

File: conf/functions_worker.yml [hunk 2]
```diff
@@ -93,6 +93,8 @@ schedulerClassName: "org.apache.pulsar.functions.worker.scheduler.RoundRobinSche
 functionAssignmentTopicName: "assignments"
 failureCheckFreqMs: 30000
 rescheduleTimeoutMs: 60000
+# frequency at which to check if cluster needs rebalancing (set to -1 to disable)
+rebalanceCheckFreqSec: 3600
```
Comment (srkukarni) [inline conf/functions_worker.yml:97]: For backwards compatibility do you want it to be -1?

**Pull request merged**